docs/shrc.md

```diff
@@ -66,43 +145,141 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
+
+:fontawesome-solid-pencil-alt:{: .note aria-hidden="true"} **Notes:**
+
+- Persistent caches are not supported on z/OS.
+- The `snapshotCache` and `restoreFromCache` cache utilities cannot be used on Windows systems.
+
+## Housekeeping
+
+Caches can be deleted if they contain many stale classes or if the cache is full and you want to create a bigger cache. Use one of the following utilities to remove unwanted caches:
+
+- [`-Xshareclasses:destroy`](xshareclasses.md#destroy): Removes specific caches when used with the `name`, `cacheDir`, and `nonpersistent` suboptions.
+- [`-Xshareclasses:destroyAll`](xshareclasses.md#destroyall): Removes all the caches that are specified by the `cacheDir` and `nonpersistent` suboptions.
+- [`-Xshareclasses:destroySnapshot`](xshareclasses.md#destroysnapshot): Removes a cache snapshot from disk that is specified by `name` and `cacheDir` suboptions.
+- [`-Xshareclasses:destroyAllSnapshots`](xshareclasses.md#destroyallsnapshots): Removes all cache snapshots from disk that are found by specifying the `cacheDir` suboption.
+- [`-Xshareclasses:destroyAllLayers`](xshareclasses.md#destroyalllayers): Removes all shared cache layers that are specified by the `name` and `cacheDir` suboptions.
+
+
+Caches can also be removed if they are unused for a specified amount of time. To configure time-based housekeeping, use the [-Xshareclasses:expire](xshareclasses.md#expire) option.
+
+If you want to remove a cache but allow it to be re-created when the VM restarts, use the [`-Xshareclasses:reset`](xshareclasses.md#reset) option.
+
 
 ## Support for custom class loaders
 
-Classes are shared by the bootstrap class loader internally in the OpenJ9 VM. If you are using a custom class loader, you can use the Java Helper API to find and store classes in the shared classes cache.
+Classes are shared by the bootstrap class loader internally in the VM. The OpenJ9 implementation of `java.net.URLClassLoader` is modified to use `SharedClassURLClasspathHelper` and any class loaders that extend `java.net.URLClassLoader` can inherit this behavior. If you are using a custom class loader that does not extend `java.net.URLClassLoader`, you can use the Java Helper API to find and store classes in a shared classes cache.
 
-For more information, see [Using the Java Helper API](https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.vm.80.doc/docs/shrc_pd_helper.html).
+If a running application uses its own class loader, you must grant the class loader permission to `SharedCachePermission` before they can share classes. To grant permission, add shared class permissions to the `java.policy` file by specifying the `ClassLoader` class name. Permissions can be set for `read`, `write`, or `read,write`.
```

Comment:
This is only required when the SecurityManager is enabled, which I thought was already fixed. Also I though there was a second paragraph about permissions, which seems to be gone now.

Reply:
I'm confused. I made some changes because of this thread, which I thought was ok? +1 https://github.com/eclipse/openj9-docs/pull/752#discussion_r604094290

Reply:
There was a sentence "If a SecurityManager is being used, ....", which I guess I accidently indicated could be removed, but I think the details where useful because they were different from the above. Possibly the "If a SecurityManager is being used" part could be removed, since the above paragraph should contain this.

Reply:
I looked back at that comment and saw that I missed adding the SecurityManager part of the agreed sentence. However, I thought you agreed to me removing the 2nd paragraph. I've put part of that back but I'll have to figure out what disappeared when I force pushed.

Reply:
Looking at the knowledgecenter I only find the following which isn't included. Not sure if this is the sentence that was removed.

If a SecurityManager is being used with class data sharing and the running application is calling the com.ibm.oti.shared.SharedClassUtilities APIs getSharedCacheInfo() or destroySharedCache(), you must grant the code calling these APIs the appropriate SharedClassesNamedPermission. For example:

Reply:
I also note there are examples in the knowledgecenter which weren't brought over. 
```
permission com.ibm.oti.shared.SharedClassPermission
        "com.abc.customclassloaders.*", "read,write";
```